solvers/NonlinearProgram.m

```diff
@@ -155,7 +155,6 @@
       obj.bbcon_lb = [];
       obj.bbcon_ub = [];
       
-      obj = obj.setSolver('default');
```

Comment:
Thanks for doing this! But I think we still want to keep the setSolver('default') here. It is better to specify a value for each property inside the constructor. If we have a derived class of `NonlinearProgram` that does not call `addConstraint`, then the solver property will not be set (which happens in `PolynomialProgram`, as the test server is complaining). I think it should work if in the next commit you put a warning and switch to fmincon. Then when you construct the problem, the default solver will be set to studentSnopt, and then when you add constraints/variables and they exceed the 300 limits, the program will switch to use fmincon automatically.